transport/memory.go

```diff
@@ -0,0 +1,60 @@
+package transport
+
+import (
+	"context"
+
+	"github.com/spacemeshos/poet/registration"
+	"github.com/spacemeshos/poet/service"
+	"github.com/spacemeshos/poet/shared"
+)
+
+type transport interface {
+	service.RegistrationService
+	registration.WorkerService
+}
+
+// inMemory is an in-memory implementation of transport.
+// It allows binding a Registration service with a Worker service
+// in a standalone mode by an in-memory channel.
+type inMemory struct {
+	closedRounds chan service.ClosedRound
+	proofs       chan shared.NIP
+}
+
+func NewInMemory() transport {
+	return &inMemory{
+		closedRounds: make(chan service.ClosedRound, 1),
+		proofs:       make(chan shared.NIP, 1),
+	}
+}
+
+// Implement registration.WorkerService.
+func (m *inMemory) ExecuteRound(ctx context.Context, epoch uint, membershipRoot []byte) error {
+	select {
+	case m.closedRounds <- service.ClosedRound{
+		Epoch:          epoch,
+		MembershipRoot: membershipRoot,
+	}:
+		return nil
+	case <-ctx.Done():
+		return ctx.Err()
+	}
+}
+
+func (m *inMemory) RegisterForProofs(ctx context.Context) <-chan shared.NIP {
```

Comment:
Yes, it supports only 1 "consumer" registered. I'd rather not complicate this until there's actually a need for it. It will probably be needed eventually for the distributed k2pow. The k2pow service will be the second consumer listening for new proofs then.